### PR TITLE
ECOM-5126 create the course-discovery setup for storage backend

### PR DESCRIPTION
**Pull request by schenedx**
@clintonb @rlucioni @e0d Please review

This change is to work with https://github.com/edx/configuration/pull/3307 to incorporate django-storage into this IDA

### CODE REVIEW

File: course_discovery/settings/production.py [hunk 1]
```diff
@@ -19,6 +19,9 @@
 # the values read from disk should UPDATE the pre-configured dicts.
 DICT_UPDATE_KEYS = ('JWT_AUTH',)
 
+# This may be overridden by the YAML in PROGRAMS_CFG, but it should be here as a default.
```
Comment (clintonb) [inline course_discovery/settings/production.py:22]: I'd prefer we be explicit about all settings, and not continue with this form of configuration.

**Comment by clintonb** [comment]
👍  (preferably with the production.py changes reverted)

File: course_discovery/settings/production.py (hunk 1 above)
Comment (rlucioni) [inline course_discovery/settings/production.py:22]: Seconded.

File: course_discovery/settings/base.py [hunk 2]
```diff
@@ -42,6 +42,7 @@
     'dry_rest_permissions',
     'compressor',
     'django_filters',
+    'storages'
```
Comment (rlucioni) [inline course_discovery/settings/base.py]: Did this turn out to be necessary?

Comment (schenedx) [inline course_discovery/settings/base.py]: It's not if we specified the value of `DEFAULT_STORAGE_BACKEND` to be within the app like Programs IDA. But we can just use the botostoragebackend provided by django-storages. And we are doing that here

**Comment by rlucioni** [comment]
👍

**Comment by schenedx** [comment]
@clintonb Can you help review again after rebased with master?

**Comment by clintonb** [comment]
👍

**Pull request merged**